src/app/static/src/app/store/root/actions.ts

```diff
@@ -51,11 +51,26 @@ export const actions: ActionTree<RootState, RootState> & RootActions = {
     },
 
     async changeLanguage(context, payload) {
-        const {dispatch, rootState} = context
-        await changeLanguage<RootState>(context, payload)
+        const {commit, dispatch, rootState} = context;
+
+        if (rootState.language === payload) {
+            return;
+        }
```

Comment:
I think this is sensible. Is the root cause that the language selector doesn't disable current language? If so then that's worthy of another ticket I think.

Reply:
I've made a ticket for this here: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2647
But with the change in this branch it is at least benign to select the current language, as nothing will happen.